internal/core/docker_host_test.go

```diff
@@ -40,24 +41,44 @@ var resetSocketOverrideFn = func() {
 	os.Setenv("TESTCONTAINERS_DOCKER_SOCKET_OVERRIDE", originalDockerSocketOverride)
 }
 
+var testCallbackCheckPassing = func(_ context.Context, _ string) error {
+	return nil
+}
+
+var testCallbackCheckError = func(_ context.Context, _ string) error {
+	return fmt.Errorf("could not check the Docker host")
+}
```

Comment:
suggestion: these don't need to be vars as they are never assigned to and functions are first class.
```suggestion
func testCallbackCheckPassing(_ context.Context, _ string) error {
	return nil
}

func testCallbackCheckError (_ context.Context, _ string) error {
	return fmt.Errorf("could not check the Docker host")
}
```